src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -514,37 +515,80 @@ public void CleanupAfterEachTest()
         }
 
         /// <summary>
-        /// Checks if the connection string provided in the config is correctly updated for MSSQL.
+        /// Checks if the connection string provided in the config is correctly updated for MSSQL by checking for the injected User Agent string w/ format:
+        /// dab_oss_Major.Minor.Patch
         /// If the connection string already contains the `Application Name` property, it should append the DataApiBuilder Application Name to the existing value.
         /// If not, it should append the property `Application Name` to the connection string.
+        /// Also validates that any custom set environment variable(e.g. hosting) for DAB_APP_NAME_ENV is used in the connection string for Application Name (or appended to the existing value).
+        /// </summary>
+        /// <param name="databaseType">database type.</param>
+        /// <param name="providedConnectionString">connection string provided in the config.</param>
+        /// <param name="expectedUpdatedConnectionString">Updated connection string with Application Name.</param>
+        /// <param name="isHostedScenario">If Dab is hosted or OSS.</param>
+        [DataTestMethod]
+        [DataRow(DatabaseType.MSSQL, "Data Source=<>;", "Data Source=<>;Application Name=", false, DisplayName = "[MSSQL]:Adding Application Name property to connectionString with dab_oss app name.")]
```

Comment:
nit: readability may be improved by moving the display name or more to new lines

Reply:
new lines in my opinion, for this test specifically, would make the datarows harder to read. Harder to distinguish between actual and expected connection strings. I updated the datarows to have more spacing to be clearer. This will be readable in Visual Studio or in CodeSpaces. Not necessarily github.
![image](https://github.com/Azure/data-api-builder/assets/6414189/93c16810-cfac-4ee3-bbf1-9528d59ffaed)